### DISCUSSION TITLE
在撰写合约时的一些疑问

### DISCUSSION THREAD

**Discussion by punkcanyang** [IntensiveCoLearning/Web3-URL#172] [General]
Ethstorage 里面如果用户删除数据，空间会释出，但是我记得Solidity里，例如机构体或是mapping都是创建后不能够删除，只能重置

Ethstorage里的删除是删除什么样的变数格式呢？

**Comment by qzhodl** [comment]
会删除存储在L2网络里面的数据以及对应的链上的数据哈希（链上将空出来的这个存储槽用来存储其他数据的哈希）

**Reply by punkcanyang** [reply]
也就是L2上重置SLOT，并将给链下储存的费用退还？

**Reply by qzhodl** [reply]
准确的说，是在L1的合约里面重置slot，且把剩下的存储费用返回给删除数据的用户
